x-pack/filebeat/input/httpjson/internal/v2/response.go

```diff
@@ -80,6 +83,10 @@ func (rp *responseProcessor) startProcessing(stdCtx context.Context, trCtx *tran
 	go func() {
 		defer close(ch)
 
+		if rp.forceNdjson {
+			resp.Header.Set("Content-Type", "application/x-ndjson")
```

Comment:
Is this just for testing purposes? If not, maybe there are other use cases for modifying response headers, and enabling writing to them in `transform_xxx.go` is a more general approach without adding specific config fields. WDYT?

Reply:
Unfortunately not just for testing, some endpoints send back ndjson with content types of text/plain or application/json.  So we need some way to force ndjson code path.  That being said I like the idea of the general approach, let me give that a try.

Reply:
After reading "if config response.force_ndjson is set, set response Content-Type
header to "application/x-ndjson" in the description, my first thought was that this sounded fairly specific. If we have several builtin handlers for different content types then maybe a generic overwrite header option would work and make future extension easier.